### PR TITLE
Database adapter documentation

### PR DESCRIPTION
**Pull request by mxstbr**
Documented not only how to use, but also how to write database adapters.

What do you think @relekang? The `subscribe` stuff is the most undetailled one because we don't yet have that functionality in core, so I don't know what we rely upon.

### CODE REVIEW

File: writing-adapters.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  put,
+  getAll,
+  has,
+  subscribe,
+}
+```
+
+Let's dive into the individual methods:
+
+### `get(key: string): Promise`
```
Comment (relekang) [inline writing-adapters.md]: Shouldn't this also accept the filtering options?

File: writing-adapters.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# Writing database adapters
+
+`micro-analytics` database adapters are simple JavaScript modules which export an object with five methods. They _have_ to be called `micro-analytics-adapter-xyz`, where `xyz` is the name users will pass to the `DB_ADAPTER` environment variable when starting `micro-analytics`.
```
Comment (sean-roberts) [inline writing-adapters.md]: I wouldn't bother specifying the number of methods anywhere. That can get out of sync quick. Probably just mention that it needs to implement the minimal interface described below.

**Comment by sean-roberts** [comment]
LGTM. Let me know if you're waiting for something on this. I am going to wait until this is merged before looking into trying a different description of the project work 👍

**Comment by mxstbr** [comment]
Nothing, just wanted to see if people like it! Need to ship a new version soon...

**Pull request merged**